app/models/user.js

```diff
@@ -389,6 +416,18 @@ exports.addModel = function(dbAdapter) {
     return this
   }
 
+  User.prototype.validateFrontendPreferencesStructure = function() {
+    // for each key in prefs there must be an object value
+    if (typeof this.frontendPreferences !== 'object')
```

Comment:
+1 for `isPlainObject`. Also, I would use `{ }` for `if` even if there's just one line. Do we use this style elsewhere in the repo?